### ISSUE TITLE
Error caused by #438 when module-info.java is present

### ISSUE THREAD

**Issue by Degubi** [eclipse-ee4j/yasson#466]
Using this code:

Main.class:
```java
public final class Main {
    public static void main(String[] args) {
        var json = JsonbBuilder.create();
        var data = """
                   {
                       "prop": "hey"
                   }
                   """;

        var deserialized = json.fromJson(data, Data.class);
    }
}
```

Data.class:
```java
public final class Data {
    public final String prop;

    @JsonbCreator
    public Data(@JsonbProperty("prop") String prop) {
        this.prop = prop;
    }
}
```

module-info.java: 
```java
open module degubi.test {
    exports degubi;

    requires jakarta.json;
    requires jakarta.json.bind;
}
```
Exception:
```
jan. 06, 2021 11:32:09 DU. org.eclipse.yasson.internal.Unmarshaller deserializeItem
SEVERE: Error accessing field 'prop' declared in 'class degubi.Data'
Exception in thread "main" jakarta.json.bind.JsonbException: Error accessing field 'prop' declared in 'class degubi.Data'
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.model.PropertyModel.createReadHandle(PropertyModel.java:550)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.model.PropertyModel.<init>(PropertyModel.java:167)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.ClassParser.lambda$parseProperties$0(ClassParser.java:70)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.HashMap$ValueSpliterator.forEachRemaining(HashMap.java:1766)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.ClassParser.parseProperties(ClassParser.java:71)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.MappingContext.lambda$createParseClassModelFunction$1(MappingContext.java:97)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.MappingContext.getOrCreateClassModel(MappingContext.java:81)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.Unmarshaller.deserializeItem(Unmarshaller.java:60)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.Unmarshaller.deserialize(Unmarshaller.java:51)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.JsonBinding.deserialize(JsonBinding.java:59)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.JsonBinding.fromJson(JsonBinding.java:66)
	at degubi.test/degubi.Main.main(Main.java:15)
Caused by: java.lang.IllegalAccessException: access to public member failed: degubi.Data.prop/java.lang.String/getField, from class org.eclipse.yasson.internal.model.PropertyModel (module org.eclipse.yasson)
	at java.base/java.lang.invoke.MemberName.makeAccessException(MemberName.java:957)
	at java.base/java.lang.invoke.MethodHandles$Lookup.checkAccess(MethodHandles.java:3625)
	at java.base/java.lang.invoke.MethodHandles$Lookup.checkField(MethodHandles.java:3575)
	at java.base/java.lang.invoke.MethodHandles$Lookup.getDirectFieldCommon(MethodHandles.java:3786)
	at java.base/java.lang.invoke.MethodHandles$Lookup.getDirectFieldNoSecurityManager(MethodHandles.java:3781)
	at java.base/java.lang.invoke.MethodHandles$Lookup.unreflectField(MethodHandles.java:3287)
	at java.base/java.lang.invoke.MethodHandles$Lookup.unreflectGetter(MethodHandles.java:3238)
	at org.eclipse.yasson@2.0.1/org.eclipse.yasson.internal.model.PropertyModel.createReadHandle(PropertyModel.java:548)
	... 18 more
```

The exception points to the unreflect calls in PropertyModel.java

I found 2 ways to fix this issue:
- Deleting the module-info.java file in my project fixes this
- I reverted back the MethodHandle changes & just used reflection in PropertyModel.java, that fixed this

Tbh, I don't fully understand why this happens...

**Comment by Degubi** [comment]
Closing this issue, because it seems that this error message is not related to the changes made in #438

**Comment by Degubi** [comment]
Fixed by #474
